### PR TITLE
Feature/#308 rocm support

### LINKED ISSUE #308: Support ROCm devices on GUI
* Support ROCm devices on GUI

## Requirements
 * [x] Check room.device tag on images
 * [x] Give users to choose rocm device allocation option
 * [ ] Let sysadmins check and modify ROCm device nodes.

### PR DESCRIPTION
**Pull request by inureyes**
This PR resolves #308

### CODE REVIEW

File: src/components/backend-ai-credential-list.ts [hunk 1]
```diff
@@ -241,7 +241,22 @@ export default class BackendAICredentialList extends BackendAIPage {
...
+          if ('rocm.device' in keypair['total_resource_slots']) {
+            keypair['total_resource_slots'].rocm_device = keypair['total_resource_slots']['rocm.device'];
+          }
+
+          if (('rocm_device' in keypair['total_resource_slots']) === false &&
+            keypair['default_for_unspecified'] === 'UNLIMITED') {
+            keypair['total_resource_slots'].rocm_device = '-';
+          }
+          if ('tpu.device' in keypair['total_resource_slots']) {
+            keypair['total_resource_slots'].rocm_device = keypair['total_resource_slots']['tpu.device'];
```
Comment (adrysn) [inline src/components/backend-ai-credential-list.ts]: `rocm_device` here should be changed to `tpu_device`?

File: src/components/backend-ai-session-list.ts [hunk 2]
```diff
@@ -1347,20 +1359,31 @@ export default class BackendAiSessionList extends BackendAIPage {
...
+                  <span class="indicator">GPU</span>
+                </template>
+                <template is="dom-if" if="[[item.tpu_slot]]">
+                  <wl-icon class="fg green indicator">view_module</wl-icon>
+                  <span>[[item.tpu_slot]]</span>
+                  <span class="indicator">TPU</span>
+                </template>
+                <template is="dom-if" if="[[!item.cuda_gpu_slot]]">
+                  <template is="dom-if" if="[[!item.cuda_fgpu_slot]]">
                     <wl-icon class="fg green indicator">view_module</wl-icon>
```
Comment (adrysn) [inline src/components/backend-ai-session-list.ts]: Perhaps ROCM gpu should also be considered here to indicate no GPUs?

```html
<template is="dom-if" if="[[!item.rocm_gpu_slot]]">
```

Comment (inureyes) [reply src/components/backend-ai-session-list.ts]: Fixed.

File: src/components/backend-ai-credential-list.ts (hunk 1 above)
Comment (inureyes) [reply src/components/backend-ai-credential-list.ts]: Fixed.

**Pull request merged**